lighthouse-core/lib/network-request.js

```diff
@@ -344,9 +357,83 @@ module.exports = class NetworkRequest {
     const totalFetchedSize = this.responseHeaders.find(item => item.name === 'X-TotalFetchedSize');
     // Bail if the header was missing.
     if (!totalFetchedSize) return;
+    const floatValue = parseFloat(totalFetchedSize.value);
+    // Bail if the header cannot be parsed
+    if (isNaN(floatValue)) return;
     this.transferSize = parseFloat(totalFetchedSize.value);
   }
 
+  /**
+   * TODO(exterkamp) add explanatory comment.
+   */
+  _updateFetchStatsForLightrider() {
+    // Bail if we somehow already have fetch stats.
+    if (!global.isLightrider) return;
+
+    // For more info on timing nomenclature: https://www.w3.org/TR/resource-timing-2/#processing-model
+
+    //    StartTime
+    //    | ConnectStart
+    //    | |     SSLStart  SSLEnd
+    //    | |     |         | ConnectEnd
+    //    | |     |         | | SendStart/End   ReceiveHeadersEnd
+    //    | |     |         | | |               |                EndTime
+    //    ▼ ▼     ▼         ▼ ▼ ▼               ▼                ▼
+    //    [ [TCP  [   SSL   ] ] [   Request   ] [   Response   ] ]
+    //    ▲ ▲     ▲         ▲ ▲ ▲             ▲ ▲              ▲ ▲
+    //    | |     '-SSLTime-' | '-requestTime-' '-responseTime-' |
+    //    | '----TCPTime------'                                  |
+    //    |                                                      |
+    //    '------------------------TotalTime---------------------'
+
+    const totalTimeHeader = this.responseHeaders.find(item => item.name === 'X-TotalTime');
+    // Bail if there was no totalTime.
+    if (!totalTimeHeader) return;
+
+    const totalTime = parseInt(totalTimeHeader.value);
+
+    const TCPTimeHeader = this.responseHeaders.find(item => item.name === 'X-TCPTime');
+    const requestTimeHeader = this.responseHeaders.find(item => item.name === 'X-RequestTime');
+    const SSLTimeHeader = this.responseHeaders.find(item => item.name === 'X-SSLTime');
+    const responseTimeHeader = this.responseHeaders.find(item => item.name === 'X-ResponseTime');
+
+    const TCPTime = TCPTimeHeader === undefined ? 0 : parseInt(TCPTimeHeader.value);
+    const requestTime = requestTimeHeader === undefined ? 0 : parseInt(requestTimeHeader.value);
+    const SSLTime = SSLTimeHeader === undefined ? 0 : parseInt(SSLTimeHeader.value);
+    const responseTime = responseTimeHeader === undefined ? 0 : parseInt(responseTimeHeader.value);
+
+    // Bail if the timings don't add up.
+    if (TCPTime + requestTime + responseTime !== totalTime) {
+      return;
+    }
+
+    // Bail if timing is not initialized.
```

Comment:
this could move to the top of the method.  

also this does expose us to some risk if LR stops including this in the response, but still has our X- headers.... but that's okay for now.